src/inmanta/logging.py

```diff
@@ -54,6 +59,102 @@ def _is_on_tty() -> bool:
 }
 
 
+class LoggingConfigExtension:
+    """
+    The logging config defined by an extension
+    """
+
+    def __init__(
+        self,
+        formatters: Optional[dict[str, object]] = None,
+        handlers: Optional[dict[str, object]] = None,
+        loggers: Optional[dict[str, object]] = None,
+        root_handlers: Optional[set[str]] = None,
+        log_dirs_to_create: Optional[set[str]] = None,
+    ) -> None:
+        """
+        :param log_dirs_to_create: The log directories that should be created before the logging config can be used.
+        """
+        self.formatters = formatters if formatters else {}
+        self.handlers = handlers if handlers else {}
+        self.loggers = loggers if loggers else {}
+        self.root_handlers = root_handlers if root_handlers else set()
+        self.log_dirs_to_create = log_dirs_to_create if log_dirs_to_create else set()
+
+    def ensure_log_dirs(self) -> None:
+        """
+        This method makes sure that the log directories, required by this logging config, are present on disk.
+        The directories are created if it doesn't exist yet.
+        """
+        for directory in self.log_dirs_to_create:
+            os.makedirs(directory, exist_ok=True)
+
+
+class LoggingConfigCore(LoggingConfigExtension):
+    """
+    The logging config defined by the inmanta-core component.
+    """
+
+    def __init__(
+        self,
+        formatters: Optional[dict[str, object]] = None,
+        handlers: Optional[dict[str, object]] = None,
+        loggers: Optional[dict[str, object]] = None,
+        root_handlers: Optional[set[str]] = None,
+        log_dirs_to_create: Optional[set[str]] = None,
+        root_log_level: Optional[int | str] = None,
+    ):
+        super().__init__(formatters, handlers, loggers, root_handlers, log_dirs_to_create)
+        self.root_log_level = root_log_level
+
+    def join(self, logging_config_extension: LoggingConfigExtension) -> "LoggingConfigCore":
```

Comment:
I had considered that name as well, but I didn't go for it because it can be confusing too. The configuration of the root logger is only a sub-part of this config. As you already mention this config contains the full logger config. Should we call it `FullLoggerConfig`?

Reply:
I understand your concern with my proposal. `FullLoggerConfig` sounds like a good compromise.